### ISSUE TITLE
Make reports and observations consecutive

### ISSUE THREAD

**Issue by JimKillock** [openrightsgroup/cmp-issues#182]
Reports and validations made for user reports and categorisations should be consecutive and dated. 

For instance, if a report is marked as "correct" or a note is added, this should be dated and then no longer editable.

New observations and notes should be added and supercede the old ones in the public front end, but the old observations should remain in the database.

This is because sites will change over time. We should retain data and observations rather than delete them.

This is the approach we settled on with copyright orders for the same reason.

**Comment by edjw** [comment]
@dantheta – Can you take a look at this when you're back from sick leave?

**Comment by dantheta** [comment]
This is already done for the categorisation operations.  It should be pretty straightforward to apply to the report review.

**Comment by dantheta** [comment]
This is done for the review notes.  The isp_report record still contains the latest value for the matches_policy field, but comments and the historical states of matches_policy are now kept in a comments append-log.

I've copied all of the previews comments from the old field to the new one, although attribution information is not available in the old schema.  I've assigned them all to Alex, hope that's OK.
